util.sh

```diff
@@ -0,0 +1,85 @@
+SLB_FILE=${SLB_FILE:=lz_header.bin}
+
+SL_SIZE=`hexdump "$SLB_FILE" -s2 -n2 -e '/2 "%u"'`
+
+validate_and_escape_hash () {
+	local TRIM=`echo -n "$1" | sed -r -e "s/ .*//"`
+	if (( ${#TRIM} != 64 && ${#TRIM} != 40 )); then
+		>&2 echo "\"$TRIM\" is not a valid SHA1/SHA256 hash"
+		return
+	fi
+	echo -n $TRIM | sed -r -e "s/([a-f0-9]{2})/\\\\\x\1/g"
```

Comment:
You can half the number of escapes for `\x` given the `-r`

Reply:
Why do I still need to use 3 backslashes, is the first escape consumed already by quotation?